lightningd/htlc_end.h

```diff
@@ -28,6 +28,14 @@ struct htlc_end {
 	u32 outgoing_cltv_value;
 	u32 cltv_expiry;
 	struct sha256 payment_hash;
+
+	/* If we are forwarding, remember the shared secret for an
+	 * eventual reply */
+	struct sha256 shared_secret;
```

Comment:
Yeah, I was a bit conflicted about re-using the `path_secrets`, I'll fix it up as a pointer.